backend/uclapi/dashboard/api_applications.py

```diff
@@ -251,12 +252,20 @@ def set_callback_url(request):
         })
         response.status_code = 400
         return response
-
-    if not is_url_safe(new_callback_url):
+    url_not_safe_saved = is_url_unsafe(new_callback_url)
+    if url_not_safe_saved:
+        if url_not_safe_saved == NOT_HTTPS:
+            message = "The requested callback URL does not "\
+                      "start with 'https://'."
+        elif url_not_safe_saved == NOT_VALID:
+            message = "The requested callback URL is not valid."
+        elif url_not_safe_saved == URL_BLACKLISTED:
+            message = "The requested callback URL is forbidden."
+        elif url_not_safe_saved == NOT_PUBLIC:
```

Comment:
Is there ever a case when none of these clauses could be called? Is it worth having a generic fall through `else` that just says that there's an unknown error with setting the callback URL?

Reply:
no other error scan occur